### ISSUE TITLE
Out-of-repo deployment doesn't work for me 

### ISSUE THREAD

**Issue by cormullion** [JuliaDocs/Documenter.jl#2403]
I'm trying to move `https://github.com/JuliaGraphics/Luxor.jl` docs to `https://github.com/JuliaGraphics/LuxorManual`. I followed the instructions in the Documenter docs [here](https://documenter.juliadocs.org/stable/man/hosting/#Out-of-repo-deployment).

The two keys were added to the correct locations, as far as I can tell, and the `make.jl` was updated with this:

```julia
repo = "github.com/JuliaGraphics/LuxorManual.git"
withenv("GITHUB_REPOSITORY" => repo) do
    deploydocs(
      repo = repo,
      target = "build",
      push_preview = true,
      forcepush = true,
    )
end
```

The error message [here](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7543161131/job/20533536894#step:7:147) says:

```
fatal: repository 'https://github.com/github.com/JuliaGraphics/LuxorManual.git.git/' not found
```

so it's possible the `make.jl` code I added is wrong, but not sure how...?

**Comment by cormullion** [comment]
Thank! Yes, the path looks better now, but although this link looks valid

https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20539088958#step:7:508

there's still an access error. Perhaps the secret keys aren't correctly set.

**Comment by goerz** [comment]
Yeah, that looks like the secrets aren’t set up right in the two repositories

**Comment by cormullion** [comment]
I’ll try and do them again tomorrow. It’s by far the worst aspect of working with Julia… 😂

**Comment by goerz** [comment]
I'd try deleting all keys and secrets from both repos and start from scratch with [`DocumenterTools.genkeys`](https://documenter.juliadocs.org/stable/man/hosting/#travis-ssh)

**Comment by cormullion** [comment]
I did the process again:

<img width="1865" alt="Screenshot 2024-01-17 at 10 10 48" src="https://github.com/JuliaDocs/Documenter.jl/assets/52289/1fae1fca-9ef9-40ea-9496-f5a723b3a8ce">

Error is:

```
remote: Permission to JuliaGraphics/LuxorManual.git denied to github-actions[bot].
[507](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20564991449#step:7:508)
fatal: unable to access 'https://github.com/JuliaGraphics/LuxorManual.git/': The requested URL returned error: 403
[508](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20564991449#step:7:509)
┌ Error: Failed to push:
[509](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20564991449#step:7:510)
│   exception =
[510](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20564991449#step:7:511)
│    failed process: Process(setenv(`/home/runner/.julia/artifacts/5d776960702ec06ebaac2a94fd96a1c44e8cb04a/bin/git push -fq upstream HEAD:gh-pages`,
```

`make.jl` code is:

```julia
repo = "JuliaGraphics/LuxorManual"
withenv("GITHUB_REPOSITORY" => repo) do
    deploydocs(
      repo = repo,
      target = "build",
      push_preview = true,
      forcepush = true,
    )
end
```

So I think the documented procedure isn't working ... ?

**Comment by cormullion** [comment]
@mortenpi  Does my screenshot above look correct? It’s my best guess based on the current documentation…

**Comment by mortenpi** [comment]
Don't think you have `DOCUMENTER_KEY` set up right, since it's trying to use `GITHUB_TOKEN`: https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20539088958#step:7:133 (also #2165).

Any reason you're trying to configure an environment, rather than just setting up a global secret? I assume the workflow just doesn't have access to the environment.. I think you have to explicitly tell a workflow to use an environment.

* https://docs.github.com/en/actions/deployment/about-deployments/deploying-with-github-actions
* https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment

**Comment by cormullion** [comment]
>Any reason you're trying to configure an environment

I have no idea 😀I think the output of `generate_keys` said to create an environment variable so I added it. Did I add it to the wrong place?

**Comment by goerz** [comment]
Should be in "Secrets and variables"

**Comment by cormullion** [comment]
>Should be in "Secrets and variables"

Thanks... I'm obviously the only one to be confused by this but I don't get it at all :( 

The Secrets tab has 3 secrets panels - the Documenter docs say "Add the DOCUMENTER_KEY secret to the "source" repository". The `genkeys()` text says "add a secure environment variable" and the Variables tab has 3 variables panels...

> rather than just setting up a global secret?

I can't see any 'global secrets? Just 'Environment Secrets', 'Repository Secrets' and 'Organisation Secrets'.

I'll try copying "the secure environment variable named 'DOCUMENTER_KEY'" to the "Actions secrets and variables/Environment secrets" panel  and see if that works.

Thanks for the help though! :)

**Comment by goerz** [comment]
I'm confused about this myself. I have the impression that the Github UI is somewhat redundant: the same "secrets" are accessible in multiple places, as far as I can tell. They've also *change* the UI over the last year or so. At some point, we should change `genkeys` to give less ambiguous instructions.

But basically, the SSH key goes into Security > Deploy Keys

![CleanShot 2024-01-26 at 13 06 40@2x](https://github.com/JuliaDocs/Documenter.jl/assets/112306/d86c20fd-b2f4-49e4-b7d1-f2c43366e685)

And the `DOCUMENTER_KEY` variable goes into Secrets and Variables > Actions > Repository Secrets

![CleanShot 2024-01-26 at 13 07 18@2x](https://github.com/JuliaDocs/Documenter.jl/assets/112306/414aaea3-f926-4365-a907-13a1d932e142)

Those screenshots are from a standard setup where the documentation of deployed to the same repository. I guess for out-of-repo deployment those two settings are just in different repositories, but otherwise it's the same. I haven't set up out-of-repo deployment myself, though.

**Comment by cormullion** [comment]
Thanks, @goerz - the pictures are worth a thousand words. I think I have set the keys and secrets correctly now. As you say I think Github must have changed the UI recently...

The [error now is ](https://github.com/JuliaGraphics/Luxor.jl/actions/runs/7544837907/job/20927374091#step:7:171):

```plain
ssh: Could not resolve hostname juliagraphics: Temporary failure in name resolution
fatal: Could not read from remote repository.
```

so I'll try again later this weekend to see if this "temporary failure" really is temporary, or is the next obstacle for me.

**Comment by goerz** [comment]
The line [`repo = "JuliaGraphics/LuxorManual"`](https://github.com/JuliaGraphics/Luxor.jl/blob/e2f7218b403e784824f2a9f860ba393dd3a7898e/docs/make.jl#L61) in your `make.jl` is wrong: that should probably be `repo = "github.com/JuliaGraphics/LuxorManual.git"`, [according to the documentation](https://documenter.juliadocs.org/stable/man/hosting/#Out-of-repo-deployment).

**Comment by cormullion** [comment]
Thanks! I’ll try that again (it didn’t work before, but perhaps it will work better with the new secrets…).

**Comment by cormullion** [comment]
It finally works. Thanks for your help!

**Comment by goerz** [comment]
My pleasure! It's the least I can do as a small Thank You for JuliaMono!
